File: backend/Syncra/src/main/java/com/edu/cit/Syncra/Entity/Job.java
package com.edu.cit.Syncra.Entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.*;

import java.util.List;

@Document(collection = "jobs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Job {

    @Id
    private String id;

    private String company;
    private String title;
    private String date;
    private List<String> tags;
    private String rate;

    // Additional fields that might be useful
    private String description;
    private String location;
    private String applicationUrl;
}

File: frontend-mobile/app/src/main/java/edu/cit/syncra/DataClass/Job.kt
package edu.cit.syncra.DataClass

data class Job(
    val id: String,
    val title: String,
    val description: String,
    val company: String,
    val location: String,
    val type: String,           // e.g., "Full-time", "Internship"
    val postedDate: String      // ISO date format or timestamp
)
